### PR TITLE
Github actions & clean up to be able to do linting

### PR DESCRIPTION
**Pull request by freddd**
### Checklist

* [X] I have read the [Contributor Guide](../../CONTRIBUTING.md)
* [X] I have read and agree to the [Code of Conduct](../../CODE_OF_CONDUCT.md)
* [X] I have added a description of my changes and why I'd like them included in the section below

### Description of Changes
This PR introduces building with Github Actions instead of using Buildkite. The goal is for this to streamline how we build open source projects. 

What does the PR contain:
* Removed the old buildkite pipeline
* Added one workflow for PR's and one for master. Master will only build and push.
* Added linting, etc using `gosec`, `golint`, `errcheck`, `staticcheck` and `go vet` to ensure that we are following best practices.
* Fixed issues raised by the linters above

What's left to do before it's ready for merge:
* **Need to add Dockerhub credentials & make it possible to push a docker image using `buildah` in the `push_master.yaml`**

This looks like a huge PR, it's actually not that big if you disregard the static assets that I decided to check in (as opposed to having them built every time). As of right now we are skipping some `staticchecks` (ST1003,U1000,ST1005) and also `golint` is run on specific files as opposed to the package due to us using `go-bindata-assetfs` to bundle the static assets in the binary. There is a PR in the `go-bindata` repo (https://github.com/go-bindata/go-bindata/pull/37) that aims to fix this and once it's merged we will start linting the whole package again. Next steps after this PR also includes building for ARM and IPv6 support, but that's for later.

For a later discussion, it's probably time to tag another release once this is merged as `latest` contains some nice fixes like following the RFC1918 specification.

Feel free to start reviewing, I only put it as draft to finish up the last thing to push to Dockerhub.

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -48,6 +48,11 @@ var (
 	filenameRe = regexp.MustCompile("[^a-zA-Z0-9]+")
 )
 
+type contextKey string
+
+const key = contextKey("úser")
```
Comment (suom1) [inline server.go]: Should it be úser and not user?

Comment (freddd) [reply server.go]: Nice catch! It should be user for sure!

**Comment by nollbit** [comment]
I'd say stop using dockerhub and instead push to Github Packages instead. That would take care of the credentials issue, no?

**Comment by suom1** [comment]
> I'd say stop using dockerhub and instead push to Github Packages instead. That would take care of the credentials issue, no?

I agree with this, let's ship packages instead!

**Comment by freddd** [comment]
Yeah, I'm currently working on it! :shipit:

**Pull request merged**